patches/countly-sdk-web+23.2.2.patch

```diff
@@ -0,0 +1,464 @@
+-                        width = Math.round(width * window.devicePixelRatio);
+-                        height = Math.round(height * window.devicePixelRatio);
++                        width = Math.round(width * globalThis.devicePixelRatio);
++                        height = Math.round(height * globalThis.devicePixelRatio);
+                     }
+                     else {
+-                        if (Math.abs(window.orientation) === 90) {
++                        if (Math.abs(globalThis.orientation) === 90) {
+                             // we have landscape orientation
+                             // switch values for all except ios
+                             var temp = width;
+@@ -3860,8 +3872,8 @@
+             }
+
+             // getting density ratio
+-            if (window.devicePixelRatio) {
+-                metrics._density = metrics._density || window.devicePixelRatio;
++            if (globalThis.devicePixelRatio) {
++                metrics._density = metrics._density || globalThis.devicePixelRatio;
+             }
+
+             // getting locale
+@@ -3873,7 +3885,7 @@
+             if (typeof document.referrer !== "undefined" && document.referrer.length) {
+                 var matches = urlParseRE.exec(document.referrer);
+                 // do not report referrers of current website
+-                if (matches && matches[11] && matches[11] !== window.location.hostname) {
++                if (matches && matches[11] && matches[11] !== globalThis.location.hostname) {
+                     var ignoring = false;
+                     if (ignoreReferrers && ignoreReferrers.length) {
+                         for (var k = 0; k < ignoreReferrers.length; k++) {
+@@ -3963,72 +3975,62 @@
+          */
+         function sendXmlHttpRequest(functionName, url, params, callback, useBroadResponseValidator) {
+             useBroadResponseValidator = useBroadResponseValidator || false;
+-            try {
+-                log(logLevelEnums.DEBUG, "Sending XML HTTP request");
+-                var xhr = null;
+-                if (window.XMLHttpRequest) {
+-                    xhr = new window.XMLHttpRequest();
+-                }
+-                else if (window.ActiveXObject) {
+-                    xhr = new window.ActiveXObject("Microsoft.XMLHTTP");
+-                }
+-                params = params || {};
+-                var data = prepareParams(params);
+-                var method = "GET";
+-                if (self.force_post || data.length >= 2000) {
+-                    method = "POST";
++            log(logLevelEnums.DEBUG, "Sending XML HTTP request");
++            params = params || {};
++            var data = prepareParams(params);
++            var method = "GET";
++            if (self.force_post || data.length >= 2000) {
++                method = "POST";
++            }
++            const fetchUrl = method === "GET" ? url + "?" + data : url;
++            const fetchReq = fetch(fetchUrl, {
```

Comment:
@whizzzkid and I discussed this on a call. XHR not available in webworker context.